### PR TITLE
Add a methlab basement, new palette for residential basements

### PR DESCRIPTION
**Pull request by anothersimulacrum**
#### Summary
```SUMMARY: Content "Adds basement with a hidden methlab"```

#### Purpose of change
The game was previously lacking a methlab basement. This aims to fix it.

#### Describe the solution
Add a few new itemgroups, a new palette, and a new basement. Two varieties, one with methlab, and one without.

#### Additional context
I might clean up the `methlab` item group and some itemgroup stuff in a few later PR, and I plan on adding some more residential basements.
Pictures:
LivepeopleLab
RetrodaysLab
ASCIILab
LivepeopleNoLab
RetrodaysNoLab
LivepeopleNoLab2
RetrodaysNoLab2

### CODE REVIEW

**Comment by Night-Pryanik** [comment]
Hiding enter to meth lab behind a bookcase makes sense, of course, but most players won't find it, ever. They have absolutely no clue that there might be something behind that bookcase. I encounterd the same problem when I designed my autodoc basement. You should add some hint - I don't know, maybe some field, or graffiti, or chems lying on the bookcase.

**Comment by Night-Pryanik** [comment]
That's a really big basement - it has two bedrooms, two bathrooms, a large dining room, a hall and several utility rooms. It looks like a completely functional home rather than a basement.

**Comment by umsretuza** [comment]
Initially I thought the screenshots were showcasing the house itself and not the basement. It resembles a surface home/2nd floor quite well. Might want to dial back on the home-like feeling, and more towards an illegal den feeling.

**Comment by Photoloss** [comment]
> That's a really big basement - it has two bedrooms, two bathrooms, a large dining room, a hall and several utility rooms. It looks like a completely functional home rather than a basement.

A rare spawn imitating European styles could make sense unless New England laws prohibit it. However the walls should be made of concrete in that case (bonus points for a red brick house on top!) and this motivation does not line up with the intent of the PR: the chance of a Europhile *also* being a DIY methhead becomes vanishingly small.

**Comment by anothersimulacrum** [comment]
Updated OP with pictures.
I removed the second bedroom, bathroom, dining room, and rec room, and added some random crate spawns.
Methlab variant will now have meth spawn on the bookshelves, toxic gas in the air and a body/knife/bleach in the laundry room to encourage investigation.

**Comment by DracoGriffin** [comment]
Is the toxic gas permanent? That'd be a bit weird.

Also, I thought the previous basement was fine (albeit too large); it's a bit cliche to always assume it wouldn't look clean. It's a bit strange to assume only rednecks or uneducated folk are somehow masterminds of the meth business.

**Comment by anothersimulacrum** [comment]
I could add back in the previous basement with a low weight. The toxic gas ins't permanent, which does bring about the problem that it might fade before you find it. The crates are just to imply that it's unfinished, after discussion on the discord.

**Comment by Mecares** [comment]
Maybe add a variant that has the same room layout as the Meth Lab but does contain normal rooms, so that the Player might recognise the Basement layout and thinks something like: That is odd normally there is the storage room in this direction instead of those shelves.

**Comment by anothersimulacrum** [comment]
Updated, and added more pictures to the OP. Added a new variant with a room behind where the bookshelves go, and added a chance for a hazmat zombie to spawn outside the room in the meth variant.
Also added chance for normal zombies to spawn in all variants, and locked the door to the laundry room in the meth variant, though it's not in the pictures.

File: data/json/mapgen/basement/basement_meth.json [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+        { "x": 18, "y": 19, "item": "chemistry_set" },
+        { "x": 11, "y": 10, "item": "towel" },
+        { "x": 16, "y": 5, "item": "bleach" },
+        { "x": 16, "y": 5, "item": "knife_steak" }
+      ],
+      "place_monster": [
+        { "monster": "mon_zombie_hazmat", "x": [ 19, 21 ], "y": [ 15, 20 ], "chance": 50 },
+        { "monster": "mon_zombie_hazmat", "x": [ 13, 14 ], "y": [ 11, 16 ], "chance": 50 },
+        { "monster": "mon_zombie_rot", "x": 15, "y": 6, "chance": 30 },
+        { "monster": "mon_zombie_common", "x": [ 13, 14 ], "y": [ 11, 17 ], "chance": 90, "repeat": [ 1, 3 ] }
```
Comment (ZhilkinSerg) [inline data/json/mapgen/basement/basement_meth.json]: There is no monster named `mon_zombie_common`. Did you mean `mon_zombie`?

File: data/json/mapgen/basement/basement_meth.json [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+        { "group": "laundry", "x": 19, "y": 6, "chance": 50, "repeat": [ 1, 10 ] },
+        { "group": "softdrugs", "x": [ 9, 10 ], "y": 12, "chance": 75, "repeat": [ 1, 3 ] },
+        { "group": "softdrugs", "x": 3, "y": 16, "chance": 75, "repeat": [ 1, 3 ] },
+        { "group": "cleaning", "x": 11, "y": 10, "chance": 90, "repeat": [ 1, 3 ] },
+        { "group": "cleaning", "x": [ 15, 16 ], "y": 5, "chance": 90, "repeat": [ 1, 3 ] },
+        { "group": "home_hw", "x": 13, "y": 20, "chance": 40, "repeat": [ 1, 3 ] },
+        { "group": "home_hw", "x": [ 21, 22 ], "y": 5, "chance": 40, "repeat": [ 1, 3 ] }
+      ],
+      "place_item": [ { "x": 11, "y": 10, "item": "towel" } ],
+      "place_monster": [ { "monster": "mon_zombie_common", "x": [ 13, 14 ], "y": [ 11, 17 ], "chance": 90, "repeat": [ 1, 3 ] } ]
```
Comment (ZhilkinSerg) [inline data/json/mapgen/basement/basement_meth.json]: There is no monster named `mon_zombie_common`. Did you mean `mon_zombie`?

File: data/json/mapgen/basement/basement_meth.json [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+        { "group": "cleaning", "x": 11, "y": 10, "chance": 90, "repeat": [ 1, 3 ] },
+        { "group": "cleaning", "x": [ 17, 18 ], "y": 16, "chance": 90 },
+        { "group": "cleaning", "x": [ 15, 16 ], "y": 5, "chance": 90, "repeat": [ 1, 3 ] },
+        { "group": "home_hw", "x": 13, "y": 20, "chance": 40, "repeat": [ 1, 3 ] },
+        { "group": "home_hw", "x": 21, "y": [ 12, 13 ], "chance": 40, "repeat": [ 1, 3 ] },
+        { "group": "home_hw", "x": [ 21, 22 ], "y": 5, "chance": 40, "repeat": [ 1, 3 ] },
+        { "group": "guns_shotgun_rare", "x": 21, "y": 14, "chance": 10 }
+      ],
+      "place_item": [ { "x": 11, "y": 10, "item": "towel" }, { "x": 21, "y": 14, "item": "wrench" } ],
+      "place_monster": [ { "monster": "mon_zombie_common", "x": [ 13, 14 ], "y": [ 11, 17 ], "chance": 90, "repeat": [ 1, 3 ] } ]
```
Comment (ZhilkinSerg) [inline data/json/mapgen/basement/basement_meth.json]: There is no monster named `mon_zombie_common`. Did you mean `mon_zombie`?

File: data/json/mapgen/basement/basement_meth.json (hunk 1 above)
Comment (anothersimulacrum) [reply data/json/mapgen/basement/basement_meth.json]: Yeah, I must have pushed old changes.

**Pull request merged**